xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -3446,11 +3457,150 @@ public void readObjectsFromForm(EditForm eform, XWikiContext context) throws XWi
+     *     "0": {
+     *       "users": ["XWiki.Admin", "XWiki.Me"],
+     *       "groups": ["XWiki.XWikiAllGroup"]
+     *     },
+     *     "1": {
+     *       "users": ["XWiki.Admin"]
+     *     }
+     *   ],
+     *   "XWiki.XWikiUsers": 
+     *     "1": {
+     *       "name": ["Spirou"]
+     *     }
+     *   ]
+     * }
+     * </pre></code>
+     * 
+     * @param request The input HTTP request that provides the parameters
+     * @return The map containing ordered data
+     */
+    private Map<String, SortedMap<Integer, Map<String, String[]>>> parseRequestUpdateOrCreate(XWikiRequest request,
+        XWikiContext context)
+    {
+        Map<String, SortedMap<Integer, Map<String, String[]>>> result = new HashMap<>();
+        @SuppressWarnings("unchecked")
+        Map<String, String[]> allParameters = request.getParameterMap();
+        for (Entry<String, String[]> parameter : allParameters.entrySet()) {
+            Matcher matcher = XPROPERTY_URL_REFERENCE.matcher(parameter.getKey());
+            if (matcher.matches() == false) {
+                continue;
+            }
+            String className;
+            String classNumberAsString;
+            Integer classNumber;
+            String classPropertyName;
+            className = matcher.group(1);
+            classNumberAsString = matcher.group(2);
+            classPropertyName = matcher.group(3);
+            DocumentReference classReference = getCurrentDocumentReferenceResolver().resolve(className);
+            try {
+                BaseClass xClass = context.getWiki().getDocument(classReference, context).getXClass();
+                if (xClass.getPropertyList().contains(classPropertyName) == false) {
+                    continue;
+                }
+                classNumber = Integer.parseInt(classNumberAsString);
+            } catch (XWikiException e) {
+                // If the class page cannot be found, skip the property update
+                LOGGER.warn("Failed to load document [{}], ignoring property update [{}]. Reason: [{}]",
+                    classReference, parameter.getKey(), ExceptionUtils.getRootCauseMessage(e));
+                continue;
+            } catch (NumberFormatException e) {
+                // If the numner isn't valid, skip the property update
+                LOGGER.warn("Invalid xobject number [{}], ignoring property update [{}].", classNumberAsString,
+                    parameter.getKey());
+                continue;
+            }
+            SortedMap<Integer, Map<String, String[]>> objectMap = result.get(className);
+            if (objectMap == null) {
+                objectMap = new TreeMap<>();
+                result.put(className, objectMap);
+            }
+            // Get the property from the right object #objectNumber of type 'objectName'; create it if they don't exist
+            Map<String, String[]> object = objectMap.get(classNumber);
+            if (objectMap.get(classNumber) == null) {
```

Comment:
You should reuse `object` instead of re-evaluating `objectMap.get(classNumber)`.